walkeeper/src/replication.rs

```diff
@@ -52,6 +54,12 @@ pub struct ReplicationConn {
     appname: Option<String>,
 }
 
+impl Drop for ReplicationConn {
+    fn drop(&mut self) {
+        let _res = self.stream_out.shutdown(Shutdown::Both);
```

Comment:
It would be nice to have a comment here explaining why this is necessary.

Reply:
done